core/templates/pages/topic-editor-page/navbar/topic-editor-navbar.component.ts

```diff
@@ -72,23 +72,25 @@ export class TopicEditorNavbarComponent {
   }
 
   _validateTopic(): void {
-    this.validationIssues = this.topic.validate();
-    if (this.topicEditorStateService.getTopicWithNameExists()) {
-      this.validationIssues.push(
-        'A topic with this name already exists.');
-    }
-    if (this.topicEditorStateService.getTopicWithUrlFragmentExists()) {
-      this.validationIssues.push(
-        'Topic URL fragment already exists.');
+    if (this.topic) {
```

Comment:
I feel like this is the wrong place to put the if/else. What causes validateTopic() to be called even if there's no topic defined? Can the if/else check go higher in the stack so that, by the time this method is called, there is definitely a topic present?

Reply:
Done, this method is called only when topic is initialized or reinitialized